woocommerce/woocommerce
# Fix/29540

### All Submissions:






### Changes proposed in this Pull Request:



Closes #29540

On the action of deleting a product category, we want to reassign any products that have no categories to a default category.

### How to test the changes in this Pull Request:

1. Create a product category.
2. Create several simple products and assign them the category from step 1.
3. Now go to the product category page and delete the category you've created in step 1.
4. Go back to the products page and ensure these products have the `Uncategorized` category assigned instead of nothing.
5. Do the same test with REST API deleting the product category via `wp-json/wc/v3/products/categories/{cat_id}` endpoint using `DELETE` method. Tip make sure to have a JSON payload of `"force": true` for this to work.

### Other information:




### Changelog entry

> Fix - Products left without default category assignment when all categories are deleted.
